server/src/main/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportUpdateDesiredNodesAction.java

```diff
@@ -140,8 +140,13 @@ static ClusterState updateDesiredNodes(ClusterState currentState, UpdateDesiredN
             }
         }
 
+        DesiredNodesMetadata.Builder desiredNodesMetadataBuilder = new DesiredNodesMetadata.Builder(proposedDesiredNodes);
```

Comment:
I think we should take into account previous desired nodes marked as members. I'll fix it.

Reply:
Tackled in https://github.com/elastic/elasticsearch/pull/84165/commits/42d504a70a8ef52a87518dae665871679048f1d9